### DISCUSSION TITLE
Allow to pass extranous files to be included in the prune task

### DISCUSSION THREAD

**Discussion by weyert** [vercel/turborepo#7464] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

I would like to suggest the ability to list a collection of files which that should copied along when pruning a project, e.g. `turbo prune --scope=demo --extra-files=.docker-ignore --extra-files=docs/**  --docker --out-dir=./docker-builds/demo`

This would make it easier to include extra files like a `.dockerignore`-file 

### Describe the solution you'd like

Allow `turbo` to copy in more files that I list when pruning a project

### Describe alternatives you've considered

Manually script to copy the file after the `prune`-task. Good alternative but still think this would  be nice to have this functionality built-in

<sub>[TURBO-1168](https://linear.app/vercel/issue/TURBO-1168/allow-to-pass-extranous-files-to-be-included-in-the-prune-task)</sub>

**Comment by arlyon** [comment]
I have added a ticket to linear and we will figure out if this makes sense for the roadmap and update here.

**Reply by weyert** [reply]
Why did this become a discussion?

**Reply by anthonyshew** [reply]
Feature requests and ideas are in Discussions now, so we have a clear place for issues to be Issues. 👍

**Comment by krinoid** [comment]
I think this is an edge-case feature but I've just hit it. I have a postinstall script in one of my packages that downloads a large binary. The postinstall script just invokes a `.js` script in `./package/scripts` directory (which skips the download when the file is present in the on-disk cache). But because I'm using `turbo prune --docker` as part of the build pipeline and I can't easily know when something that depends on my package is getting build, I can't easily manually copy this file. Would be very helpful to have a standard/easy way for defining what should be included in the prunned output.

Maybe this could be a part of the `turbo.json` config? E.g. `includeInPrune: ["/some_pattern/"]` or something like that?

**Reply by krinoid** [reply]
I've explored the prune command code: https://github.com/vercel/turborepo/blob/6c85c5aec1fe6c4d65d1534a58fc5af00a643047/crates/turborepo-lib/src/commands/prune.rs#L67 and I see that there are internal concepts of `ADDITIONAL_DIRECTORIES` & `ADDITIONAL_FILES`. I think that if these could be exposed as configuration/CLI flags it might solve that problem.

If this feature request is accepted I'm happy to contribute/implement it, as I know Rust (albeit on an intermediate level). I'd probably need a bit of guidance/suggestions but I guess this can be done as part of the PR. Let me know what you think, thanks!

**Reply by krinoid** [reply]
Attempted implementation: https://github.com/vercel/turborepo/pull/10989

**Comment by theogravity** [comment]
It would be great to not have to add additional copy commands in the dockerfile if I can specify the additional files I require during a prune operation.
